src/code/z_camera_data.c

```diff
@@ -59,3188 +85,4737 @@ char sCameraModeNames[][12] = {
-    { 300, 2 },    // distMax
-    { 20, 3 },     // pitchTarget
-    { 15, 4 },     // swingUpdateRate
-    { 5, 5 },      // unk_10
-    { 40, 6 },     // unk_14
-    { 60, 7 },     // fovTarget
-    { 100, 8 },    // unk_1C
-    { 0x0004, 9 }, // flags
-};
-
-CameraModeValue D_8011AC08[] = {
-    /* CAM_FUNC_PARA1 */
-    { -50, 0 },    // unk_00
-    { 250, 1 },    // distTarget
-    { 70, 3 },     // pitchTarget
-    { 0, 10 },     // yawTarget
-    { 15, 4 },     // unk_08
-    { 5, 5 },      // unk_0C
-    { 60, 7 },     // fovTarget
-    { 100, 8 },    // unk_14
-    { 0x200A, 9 }, // flags
-    { -50, 11 },   // unk_18
-    { 20, 12 },    // unk_1C
-};
-
-CameraModeValue D_8011AC34[] = {
-    /* CAM_FUNC_KEEP3 */
-    { -30, 0 },    // yOffset
-    { 70, 1 },     // minDist
-    { 200, 2 },    // maxDist
-    { 40, 13 },    // swingYawInitial
-    { 10, 14 },    // swingYawFinal
-    { 10, 15 },    // swingPitchInitial
-    { 20, 16 },    // swingPitchFinal
-    { 70, 17 },    // swingPitchAdj
-    { 45, 7 },     // fovTarget
-    { 10, 8 },     // atLERPScaleMax
-    { 10, 4 },     // initTimer
-    { 0x3500, 9 }, // flags
-};
-
-CameraModeValue sSetNorm3ModeBoomVals[] = {
-    /* CAM_FUNC_JUMP3 */
-    /* mode is BOOMERANG, but these values are for when the eye
+/**
+ *=====================================================================
+ *                   General Data: NORMAL0 Setting
+ *=====================================================================
+ */
+
+/**
+ * Setting    Mode      Function
+ * -------    ------    --------
+ * NORMAL0    NORMAL    NORM1
+ */
+CameraModeValue sSetNormal0ModeNormalData[] = {
+    { -20, CAM_DATA_Y_OFFSET },
+    { 200, CAM_DATA_EYE_DIST },
+    { 300, CAM_DATA_EYE_DIST_NEXT },
+    { 10, CAM_DATA_PITCH_TARGET },
+    { 12, CAM_DATA_YAW_UPDATE_RATE_TARGET },
+    { 10, CAM_DATA_XZ_UPDATE_RATE_TARGET },
+    { 35, CAM_DATA_MAX_YAW_UPDATE },
+    { 60, CAM_DATA_FOV },
+    { 60, CAM_DATA_AT_LERP_STEP_SCALE },
+    { 0x0003, CAM_DATA_FLAGS },
+};
+
+/**
+ * Setting                 Mode      Function
+ * --------------------    ------    --------
+ * NORMAL0                 TARGET    PARA1
+ * BOSS_GOHMA              TARGET    PARA1
+ * BOSS_DODONGO            TARGET    PARA1
+ * BOSS_BARINADE           TARGET    PARA1
+ * BOSS_PHANTOM_GANON      TARGET    PARA1
+ * BOSS_VOLVAGIA           TARGET    PARA1
+ * BOSS_BONGO              TARGET    PARA1
+ * BOSS_MORPHA             TARGET    PARA1
+ * BOSS_TWINROVA_PLATFORM  TARGET    PARA1
+ * BOSS_TWINROVA_FLOOR     TARGET    PARA1
+ * BOSS_GANONDORF          TARGET    PARA1
+ * BOSS_GANON              TARGET    PARA1
+ * TOWER_CLIMB             TARGET    PARA1
+ * TOWER_UNUSED            TARGET    PARA1
+ * PIVOT_IN_FRONT          TARGET    PARA1
+ * NORMAL2                 TARGET    PARA1
+ * JABU_TENTACLE           TARGET    PARA1
+ * DIRECTED_YAW            TARGET    PARA1
+ * NORMAL4                 TARGET    PARA1
```

Comment:
The information in these comments seems backwards. It looks the data being referenced in the settings below is what should indicate the modes/functions they are used in, and there is no need to couple this information both ways.

Generally speaking, I think comments like this should be kept to the mininum otherwise it's going to be a nightmare to mod while keeping comments accurate.

Reply:
I'll agree that the thoughts behind these comments were more of a making everything as clearly documented as possible instead of a modding context, so any suggestion trying to organize this file that accomplishes both would be great. In the end, this data is basically organized into a 2D table, where each combination of setting and mode gets associated with both a camera function and a set of data used by that camera function, and I'd like that clearly documented somewhere that's easy to read. 

Having these comments above the data was a first attempt which I think is better than nothing, and was also an attempt to make this file more navigable. I'm not sure the names along are enough to make this file readable, but any alternatives bloats the file with comments as you've pointed out. So I'm not entirely sure what the best solution is moving forward. This is more of a first step though to get a discussion rolling if possible

Reply:
I wasn't referring to any names with this comment, to be clear. To me all the information here is already provided by which setting uses the data along with the corresponding mode and function enums next to the reference in the setting, so these comments don't add anything and just couple the information for no real gain imo.

Reply:
I think the tables are good to have. There's no way one would get a clear view of the 10-20 uses of `CameraModeValue[]` data without looking at the uses side by side, which this table achieves.

So I think the information is good to have synthesized. However I'd be personally open to formatting it differently, for example:

```c
/**
 * Setting                 Mode      Function
 * --------------------    ------    --------
 * NORMAL0                 TARGET    PARA1
 * BOSS_GOHMA              TARGET    PARA1
 * BOSS_DODONGO            TARGET    PARA1
 * BOSS_BARINADE           TARGET    PARA1
 * BOSS_PHANTOM_GANON      TARGET    PARA1
 * BOSS_VOLVAGIA           TARGET    PARA1
 * BOSS_BONGO              TARGET    PARA1
 * BOSS_MORPHA             TARGET    PARA1
 * BOSS_TWINROVA_PLATFORM  TARGET    PARA1
 * BOSS_TWINROVA_FLOOR     TARGET    PARA1
 * BOSS_GANONDORF          TARGET    PARA1
 * BOSS_GANON              TARGET    PARA1
 * TOWER_CLIMB             TARGET    PARA1
 * TOWER_UNUSED            TARGET    PARA1
 * PIVOT_IN_FRONT          TARGET    PARA1
 * NORMAL2                 TARGET    PARA1
 * JABU_TENTACLE           TARGET    PARA1
 * DIRECTED_YAW            TARGET    PARA1
 * NORMAL4                 TARGET    PARA1
 */
CameraModeValue sSetNormal0ModeTargetData[] = {
```
repeating `TARGET`/`PARA1` could be avoided by turning it into
```c
/**
 * Used by 19 Settings:
 * NORMAL0, BOSS_GOHMA, BOSS_DODONGO, BOSS_BARINADE, BOSS_PHANTOM_GANON, BOSS_VOLVAGIA, BOSS_BONGO, BOSS_MORPHA,
 * BOSS_TWINROVA_PLATFORM, BOSS_TWINROVA_FLOOR, BOSS_GANONDORF, BOSS_GANON, TOWER_CLIMB, TOWER_UNUSED, PIVOT_IN_FRONT,
 * NORMAL2, JABU_TENTACLE, DIRECTED_YAW, NORMAL4
 * 
 * With mode:     TARGET
 * With function: PARA1
 */
CameraModeValue sSetNormal0ModeTargetData[] = {
```

Reply:
Alright, so I've actually decided to try a different approach when it comes to this data (and I can always go back to the previous style if need be thanks to git).

But one constraint when working with this data is that it has to match up with the function that uses the data. The current style didn't capture that at all, but now that info is contained in macros. And a lot of info of what each piece of data does was basically moved from everywhere in camera_data to the macro definition of a particular function's data, which is placed next to the corresponding dataStruct for that function.

With the mapping of settings and modes near the end of the data cleaned up and easier to read, I think it's easier to try and look up which settings use a particular set of data, instead of having everything in the table. And the setting and mode are captured in the data name. And the function its used in is captured in the macro name. I'd be curious on both your thoughts @Roman971 @Dragorn421 on this newer approach